File: MatchMe-tvOSGameObjC Cropped/MatchMe/MatchMe/Configurations.m
//
//  Configurations.m
//  MatchMe
//
//  Created by Siyana Slavova on 2/22/16.
//  Copyright © 2016 mycompany.MatchMe. All rights reserved.
//

#import "Configurations.h"

@implementation Configurations

NSInteger const maxRowsCount = 3;
NSInteger const maxColumnCount = 6;

NSInteger const easyLevelCardCount = 8;
NSInteger const normalLevelCardCount = 12;
NSInteger const hardLevelCardCount = 18;

NSString* const kCurrentUserNameKey = @"CurrentUserNameKey";
NSString* const kBestPlayerNameKey = @"BestPlayerNameKey";
NSString* const kBestPlayerTimeKey = @"BestPlayerTimeKey";
NSString* const kBestPlayerTriesKey = @"BestPlayerTriesKey";

NSString* const kEasyLevelKey = @"EasyLevelKey";
NSString* const kNormalLevelKey = @"NormalLevelKey";
NSString* const kHardLevelKey = @"HardLevelKey";

+ (NSString *)nameForLevelMode:(LevelMode)levelMode {
    NSString *result = @"";
    switch (levelMode) {
        case LevelModeEasy: {
            result = @"easy level";
            break;
        }
        case LevelModeNormal: {
            result = @"normal level";
            break;
        }
        case LevelModeHard: {
            result = @"hard level";
            break;
        }
        case LevelsCount: {
            
            break;
        }
    }
    return result;
}

@end

File: MatchMe-tvOSGame Cropped/MatchMe/Configurations.swift
//
//  Configurations.swift
//  MatchMe
//
//  Copyright © 2015. All rights reserved.
//

import UIKit

let levelsNumber : NSInteger! = 3
var levelRowNumber : NSInteger!
var levelColumnNumber : NSInteger!

let maxRowsCount : NSInteger! = 3
let maxColumnCount : NSInteger! = 6

var easyLevelCardCount : NSInteger! = 8
let normalLevelCardCount : NSInteger! = 12
let hardLevelCardCount : NSInteger! = 18

enum Level: Int {
    case Easy
    case Normal
    case Hard
    static var count: Int {
        return Level.Hard.hashValue + 1
    }
}

struct Card {
    var name : String
    var image : UIImage
}

let kCurrentUserNameKey : String! = "CurrentUserNameKey"
let kBestPlayerNameKey : String! = "BestPlayerNameKey"
let kBestPlayerTimeKey : String! = "BestPlayerTimeKey"
let kBestPlayerTriesKey : String! = "BestPlayerTriesKey"

let kEasyLevelKey : String! = "EasyLevelKey"
let kNormalLevelKey : String! = "NormalLevelKey"
let kHardLevelKey : String! = "HardLevelKey"
